### PR TITLE
allow RPMs to install

### PR DESCRIPTION
**Pull request by d-w-moore**
This, again, is the necessary CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION to prevent RPM conflicts between the plugin package and the base iRODS packages for paths under `/var/lib/irods` .

### CODE REVIEW

**Comment by d-w-moore** [comment]
This has been necessary also for [curl](https://github.com/irods/irods_microservice_plugins_curl/pull/49) and [logical quotas](https://github.com/irods/irods_rule_engine_plugin_logical_quotas/pull/73) plugins.  In all cases CentOS-like OS'es always fail to reach the tests without this CMakeList fix.  But I find this  strange: Wouldn't the tests have been failing for our previous releases as well? 
Either:
  1. we did set the ...EXCLUDE... variable in 4-2-stable but didn't cherry-pick to main (**for which I see no evidence**).
  1. our Jenkins testing always installed plugins with an RPM force option, or
  2. we're setting an option for the RPM packages, or ...
  3. there's something else I'm missing.
@alanking @SwooshyCueb  @trel

File: CMakeLists.txt [hunk 1]
```diff
@@ -72,6 +72,16 @@ endif()
...
 
+if (NOT CPACK_PACKAGING_INSTALL_PREFIX)
+  set(CPACK_PACKAGING_INSTALL_PREFIX "/" CACHE STRING "Package root path. \"/\" is correct for normal package builds.." FORCE)
+  message(STATUS "Setting unspecified CPACK_PACKAGING_INSTALL_PREFIX to '${CPACK_PACKAGING_INSTALL_PREFIX}'. This is the correct setting for normal builds.")
+endif()
+
+list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "${CPACK_PACKAGING_INSTALL_PREFIX}var/lib/irods")
+list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "${CPACK_PACKAGING_INSTALL_PREFIX}var/lib/irods/scripts")
+list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "${CPACK_PACKAGING_INSTALL_PREFIX}var/lib/irods/scripts/irods")
+list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "${CPACK_PACKAGING_INSTALL_PREFIX}var/lib/irods/scripts/irods/test")
```
Comment (korydraughn) [inline CMakeLists.txt:83]: Don't forget to use the recommendation provided by @SwooshyCueb as seen here.
https://github.com/irods/irods_rule_engine_plugin_python/blob/8f7356e499e60a4e55475506b0078a02d115f842/CMakeLists.txt#L126-L131

Notice, the use of `include(IrodsCPackCommon)` as well as the `list(APPEND ...)` lines.

**Review by korydraughn — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (d-w-moore) [reply CMakeLists.txt:83]: the tiering CMakeLists.txt already has such an include, but in a different order. I may be missing the sublety here. @SwooshyCueb

Comment (alanking) [reply CMakeLists.txt:83]: `var/lib/irods` -> `${IRODS_HOME_DIRECTORY}` in all cases

Comment (d-w-moore) [reply CMakeLists.txt:83]: @alanking  - Does that mean it should read as follows?:
```list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "${CPACK_PACKAGING_INSTALL_PREFIX}${IRODS_HOME_DIRECTORY}")
```

Comment (d-w-moore) [reply CMakeLists.txt]: Done .

**Comment by d-w-moore** [comment]
Seems to pass on ubuntu 18 now that the ireg option is updated from -C to -r in that one test. Going back to Centos7 again for another attempt.

File: CMakeLists.txt [hunk 2]
```diff
@@ -74,6 +74,16 @@ unset(CPACK_DEBIAN_PACKAGE_VERSION CACHE)
 
 include(IrodsCPackCommon)
 
+if (NOT CPACK_PACKAGING_INSTALL_PREFIX)
+  set(CPACK_PACKAGING_INSTALL_PREFIX "/" CACHE STRING "Package root path. \"/\" is correct for normal package builds.." FORCE)
+  message(STATUS "Setting unspecified CPACK_PACKAGING_INSTALL_PREFIX to '${CPACK_PACKAGING_INSTALL_PREFIX}'. This is the correct setting for normal builds.")
+endif()
```
Comment (alanking) [inline CMakeLists.txt:80]: Sounds like we are trying to confirm whether this block is necessary in the other PRs, which would similarly affect this one, right?

Comment (korydraughn) [reply CMakeLists.txt:80]: That is correct.

**Comment by d-w-moore** [comment]
squashed/#'d

**Review by alanking — APPROVED** [review]

**Pull request merged**